pkg/webhook/v1beta1/experiment/validator/validator_test.go

```diff
@@ -570,6 +572,16 @@ spec:
 			Err:             false,
 			testDescription: "Trial template contains Trial parameters when spec.parameters is empty",
 		},
+		// Trial template contains Trial metadata parameter substitution
+		{
+			Instance: func() *experimentsv1beta1.Experiment {
+				i := newFakeInstance()
+				i.Spec.TrialTemplate.TrialParameters[1].Reference = "${trialSpec.Name}"
+				return i
+			}(),
+			Err:             false,
+			testDescription: "Trial template contains Trial metadata reference as parameter",
+		},
```

Comment:
Could you add a test case for `Labels` or `Annotations`?

Reply:
hi @tenzen-y , I've added handing of those cases and tests, thanks!